packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -651,14 +704,17 @@ class ClipRRectLayer extends ContainerLayer {
   ///
   /// The [clipRRect] property must be non-null before the compositing phase of
   /// the pipeline.
-  ClipRRectLayer({ this.clipRRect });
+  ClipRRectLayer({ this.clipRRect, this.clipBehavior = Clip.antiAlias });
 
   /// The rounded-rect to clip in the parent's coordinate system.
   ///
   /// The scene must be explicitly recomposited after this property is changed
   /// (as described at [Layer]).
   RRect clipRRect;
 
+  /// {@macro flutter.layer.clipBehavior}
+  Clip clipBehavior;
```

Comment:
this says it's not allowed to be `Clip.none`, which seems reasonable, but you don't assert it above (which you do in other constructors) and you do support it in `addToScene` (which you don't in the others), which seems confusing. I think they should all be consistent (and probably not support none).

Reply:
It seems weird that you support it in addToScene and indeed have a mutable field that can be set to `none` but don't allow it in the constructor. I hadn't realized it was a mutable field before (it makes sense that it is). Maybe the field should have a setter that asserts it's valid?

as in:

```dart
  Clip get clipBehavior => _clipBehavior;
  Clip _clipBehavior;
  set clipBehavior(Clip value) {
    assert(value != null);
    assert(value != Clip.none);
    _clipBehavior = value;
  }
```

Reply:
if you did that you could remove the behaviour in addToScene that cares about none.